code/modules/projectiles/guns/energy.dm

```diff
@@ -116,17 +121,21 @@
 		update_wear_icon()
 
 /obj/item/weapon/gun/energy/MouseDrop(over_object)
-	if(!self_recharge)
-		if((src.loc == usr) && istype(over_object, /obj/screen/inventory/hand) && eject_item(cell, usr))
-			cell = null
-			update_icon()
-	else
+	if(disposable)
+		to_chat(usr, SPAN_WARNING("[src] is a disposable, its batteries cannot be removed!."))
+	if(self_recharge)
 		to_chat(usr, SPAN_WARNING("[src] is a self-charging gun, its batteries cannot be removed!."))
+	if((src.loc == usr) && istype(over_object, /obj/screen/inventory/hand) && eject_item(cell, usr))
+		cell = null
+		update_icon()
```

Comment:
There's a issue with it, when the gun is dragged to hands, the code dosen't prevent from pulling the cell, which is why if...else if....else block was used to do that.